File: tests/fixtures/graph-src-tier2/go/geometry.go
// Package shapes provides geometry primitives for the Go tier-2 fixture.
//
// Exercises: package-level func, method (receiver), struct + fields, type
// alias, package-level variable, imports, and cross-file calls.
package shapes

import "math"

// Scalar is a type alias over a primitive.
type Scalar = float64

// Point is a struct with named fields (struct members).
type Point struct {
	X Scalar
	Y Scalar
}

// Unit is a package-level variable.
var Unit Scalar = 1.0

// Hypot is a free function used by DistanceTo and across files.
func Hypot(a, b Scalar) Scalar {
	return math.Sqrt(a*a + b*b)
}

// DistanceTo is a method with a value receiver (intra-file call).
func (p Point) DistanceTo(other Point) Scalar {
	dx := p.X - other.X
	dy := p.Y - other.Y
	return Hypot(dx, dy)
}

File: tests/fixtures/graph-src-tier2/java/Geometry.java
package fixture;

/** Geometry helpers for the Java tier-2 fixture. */
public class Geometry {
    /** A class-level constant field. */
    public static final double UNIT = 1.0;

    /** Static free function used by Point.distanceTo and across files. */
    public static double hypot(double a, double b) {
        return Math.sqrt(a * a + b * b);
    }
}
